src/Background/estimators.jl

```diff
@@ -61,3 +61,50 @@ julia> estimate_background(Mode, data)
 struct Mode <: BackgroundEstimator end
 
 estimate_background(::Mode, data; dims = :) = mode(data)
+
+"""
+    SourceExtractor
+
+This estimator returns the background of the input using the SourceExtractor algorithm.
+
+The background is calculated using a mode estimator of the form `(2.5 * median) - (1.5 * mean)`.
+
+If `(mean - median) / std > 0.3` then the median is used and if `std = 0` then the mean is used.
+
+# Example
+```jldoctest
+julia> data = ones(5 ,5);
+
+julia> estimate_background(SourceExtractor, data)
+1.0
+
+julia> estimate_background(SourceExtractor, data, dims=1)
+1×5 Array{Float64,2}:
+ 1.0  1.0  1.0  1.0  1.0
+```
+"""
+struct SourceExtractor <: BackgroundEstimator end
+
+function estimate_background(::SourceExtractor, data; dims = :)
+    _mean = mean(data, dims = dims)
+    _median = median(data, dims = dims)
+    _std = std(data, dims = dims)
+
+    background = 2.5 * _median - 1.5 * _mean
+    if !(background isa Float64)
+        for i in length(background)
+            if _std[i] == 0
+                background[i] = _mean[i]
+            elseif abs(_mean[i] - _median[i])/_std[i] > 0.3
+                background[i] = _median[i]
+            end
+        end
+    else
+        if _std == 0
+            background = _mean
+        elseif abs(_mean - _median)/_std > 0.3
+            background = _median
+        end
+    end
+    return background
+end
```

Comment:
Hm, this seems overly complicated because of the manual distinctions between arrays and scalars. We don't necessarily have to do that. I'll start off with 
```julia
background = @. 2.5 * _median - 1.5 * _mean
```
even if `_median` and `_mean` are scalars, broadcasting will no-op and we get scalar mathematics. 

Next, we do the comparisons. I would recommend the following:
```julia
function validate_SE(background::Number, _mean::Number, _median::Number, _std::Number)
    _std \approx 0 && return _mean
    abs(_mean - _median) / _std > 0.3 && return _median
    return background
end
```

then, from within the main function, call
```julia
return validate_SE.(background, _mean, _median, _std)
```

Reply:
I haven't tested if that works exactly, but I think it should :)